m3u8/model.py

```diff
@@ -1119,6 +1137,57 @@ def dumps(self):
     def __str__(self):
         return self.dumps()
 
+class DateRangeList(TagList):
+    pass
+
+class DateRange(object):
+    def __init__(self, **kwargs):
+        self.id = kwargs.get('id')
+        self.start_date = kwargs.get('start_date')
+        self.class_ = kwargs.get('class')
+        self.end_date = kwargs.get('end_date')
+        self.duration = kwargs.get('duration')
+        self.planned_duration = kwargs.get('planned_duration')
+        self.scte35_cmd = kwargs.get('scte35_cmd')
+        self.scte35_out = kwargs.get('scte35_out')
+        self.scte35_in = kwargs.get('scte35_in')
+        self.end_on_next = kwargs.get('end_on_next')
+        self.x_client_attrs = [ (attr, kwargs.get(attr)) for attr in kwargs if attr.startswith('x_') ]
+
+    def dumps(self):
+        daterange = []
+        daterange.append('ID=' + quoted(self.id))
+
+        if (self.start_date):
+            daterange.append('START-DATE=' + quoted(self.start_date))
+        if (self.class_):
+            daterange.append('CLASS=' + quoted(self.class_))
+        if (self.end_date):
+            daterange.append('END-DATE=' + quoted(self.end_date))
+        if (self.duration):
+            daterange.append('DURATION=' + int_or_float_to_string(self.duration))
+        if (self.planned_duration):
+            daterange.append('PLANNED-DURATION=' + int_or_float_to_string(self.planned_duration))
+        if (self.scte35_cmd):
+            daterange.append('SCTE35-CMD=' + self.scte35_cmd)
+        if (self.scte35_out):
+            daterange.append('SCTE35-OUT=' + self.scte35_out)
+        if (self.scte35_in):
+            daterange.append('SCTE35-IN=' + self.scte35_in)
+        if (self.end_on_next):
+            daterange.append('END-ON-NEXT=' + self.end_on_next)
+
+        for attr, value in self.x_client_attrs:
+            daterange.append('%s=%s' % (
+                    denormalize_attribute(attr),
```

Comment:
This indentation does not look right, does it?

Reply:
It doesn't look right. It seems to work for reasons I don't understand, but i'll correct it anyway.

Reply:
Indentation fixed via 22bce1e.